swupd/files.go

```diff
@@ -271,11 +271,18 @@ func (f *File) findFileNameInSlice(fs []*File) *File {
 	return nil
 }
 
-func (f *File) findIManifestInSlice(fs []*File) *File {
-	// get bundle name without IManifest version
-	prefix := strings.SplitAfter(f.Name, ".I.")[0]
-	for _, file := range fs {
-		if strings.HasPrefix(file.Name, prefix) {
+func (f *File) findBundleInManifest(manifest *Manifest) *File {
+	if f.Type != TypeManifest && f.Type != TypeIManifest {
+		return nil
+	}
+
+	name_split := func(str string) string {
+		return strings.SplitAfter(str, ".I.")[0]
+	}
+
+	name := name_split(f.Name)
+	for _, file := range manifest.Files {
+		if name == name_split(file.Name) {
```

Comment:
I think the original is much more performant just checking the prefix and not doing a `SplitAfter()` on _every_ bundle.